src/components/astra-editor/index.ts

```diff
@@ -317,4 +351,12 @@ export class AstraEditor extends LitElement {
       }
     }
   }
+
+  addStyle(name: string, styles: string) {
+    this.styleSheets = [...this.styleSheets, { name, styles }]
+  }
+
+  removeStyle(name: string) {
+    this.styleSheets = this.styleSheets.filter((s) => s.name !== name)
+  }
```

Comment:
could we modify the `styles` from line 37 instead of injecting styles tags into the components markup?
I'm not really following why a component needs to receive styles as an attribute at all; is this for injecting themes? if so, could we use CSS variables instead that web components natively have access to?

Reply:
The `styleSheets` is internal state, it is not an attribute. 
- `addStyle` allows plugin to inject custom style
- `removeStyle` for clean up their custom style when it is removed

> if so, could we use CSS variables instead that web components natively have access to?

It is more than what CSS variable can do. For example, I need to inject this class to editor https://github.com/outerbase/astra-ui/blob/ca69e3f0374e8a6ae6a60d0c38664629479c36e8/src/components/astra-editor/plugins/prompt.ts#L403-L408

This class is used for highlight the line of code returned by ezql.

> could we modify the styles from line 37

We could, but then for every new plugin that need to inject the style, we need to modify the the editor styles. I want the styles needed for plugin to be inside the plugin.